### DISCUSSION TITLE
Question on DAC handling

### DISCUSSION THREAD

**Discussion by robmaxtech** [SpenceKonde/DxCore#98] [Q&A] [answered]
First, thanks a lot for this brilliant library - again, like all of your stuff, now for this exciting chips too!
I encountered an issue writing the DAC by
DAC0.DATA = myvalue;

This does nothing; but when writing DAC0.DATAL and DATAH, then the analog output changes as expected.
(Intitalizing by setting DAC_ENABLE and DAC_OUTEN is done, of course).
Am I missing something here with writing 16bit-registers? With another one, e.g. CCMP in TCB0, it works fine..
Thanks for a hint!

**Comment by SpenceKonde** [comment, marked as answer]
hm, are you shifting the data appropriately? if you have a number X that is a 10 bit desired output value, you're setting DAC0.DATA = X <<6? That register is left-adjusted so it can be a normal 16-bit register while letting you write the low byrte onlyto use as an 8-bit DAC.,

**Reply by robmaxtech** [reply]
DAMNED! You are right of couurse; I should have read the datasheet more carefully... Thanks a lot!

**Comment by SpenceKonde** [comment]
This code works for me:

```c
void setup() {
  // put your setup code here, to run once:
  Serial.begin(115200);
  Serial.println("reset");
  delay(4000); 
  DACReference(VDD);
  DAC0.CTRLA=DAC_ENABLE_bm | DAC_OUTEN_bm;
  DAC0.DATA=(uint16_t)0x8000;
  analogRead(ADC_DAC0); //dummy read  - first analog read I take (of anything, not just this) comes back 0, investigating.
}

void loop() {
  // put your main code here, to run repeatedly:
  static uint16_t dacval = 0x110;
  DAC0.DATA = (dacval << 6);
  delay(1000);
  Serial.printHex(dacval);
  Serial.print(" ");
  Serial.printHex(DAC0.DATA);
  Serial.print(" ");
  Serial.println(analogRead(ADC_DAC0));
  dacval+= 0x20;
  dacval &=0x03FF;
}
```
